src/pages/PickPage.jsx

```diff
@@ -48,6 +53,12 @@ function PickPage() {
 
       if (isDuplicatedFoodId(foodId, list)) return;
       setter([...list, { id: foodId, name: foodName, img: foodImg }]);
+
+      if (list.length > 4) {
+        setRestrictModal(true);
+        setCheckType(list === coEatList ? '코잇' : '노잇');
+        setter(list.filter((el, idx) => idx < list.length));
+      }
```

Comment:
```suggestion
      if (list.length >= 5) {
        setRestrictModal(true);
        setCheckType(list === coEatList ? '코잇' : '노잇');
        return;
      }
      setter([...list, { id: foodId, name: foodName, img: foodImg }]);
```

추가한 뒤에 조건을 검사해서 다시 지우는 것보다는 미리 조건을 검사하고 추가시키지 않는 게 더 좋을 것 같습니다!

Reply:
setter가 if문 바깥에 있으면 무조건 추가되는 것이 아닌가요?

Reply:
> setter가 if문 바깥에 있으면 무조건 추가되는 것이 아닌가요?


만약 이미 코잇/노잇이 5개 이상이면 모달을 띄우고 58번 라인에서 return 함으로 밑의 라인들은 실행하지 않을 것 같아요

Reply:
아하 알겠습니다!!